### ISSUE TITLE
Support for sending TOTP tokens via NumLock/CapsLock/ScrollLock

### ISSUE THREAD

**Issue by cpfeiffer** [Nitrokey/nitrokey-app#10]
At the moment it is possible to send HOTP tokens by pressing NumLock/CapsLock or ScrollLock, as configured in the settings dialog.

It would be nice if one could also/instead send TOTP tokens that way.

**Comment by jans23** [comment]
The key combination to trigger HOTP works even without the Nitrokey App. A similar feature for TOTP would still require the Nitrokey App being running (because it needs to send the time to the device). Would this still be useful for you?

The feature would implicate to assign key combinations to TOTP slots in the OTP config dialog.

Would you be interested to help and provide a patch for this?

**Comment by cpfeiffer** [comment]
Good point about the app needed to be running. Would this work like this
1. key recognizes NumLock input
2. key magically tells the app that a TOTP token is requested
3. app sends the command with the timestamp to the key
4. key sends the token

I'm afraid that 2. might not be possible.
Alternatively it could work like this:
1. app recognizes NumLock input (or other shortcut)
2. app sends the command with the timestamp to the key
3. key sends the token

But then the app would need to register some kind of hook/global shortcut to get invoked even when not having the focus.

I could try to help out with, at least with the app part.

**Comment by jans23** [comment]
The 2nd option should be the way to go. I'm looking forward to see your patch. :-)

**Comment by jans23** [comment]
The aim of the "NumLock feature" is to enable the usage of Nitrokey OTP **without** the NK App. Your suggestion would require the NK App (of course). What would be its benefit?
